spec/services/facility_stats_service_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'rails_helper'
+
+RSpec.describe FacilityStatsService do
+  let(:today) {'January 15th 2021' }
+  let(:four_months_ago) { 'September 20th 2020' }
+  let(:five_months_ago) { 'August 15th 2020' }
+  let(:six_months_ago) { 'July 25th 2020' }
+
+  let(:organization) { create(:organization, name: "org-1") }
+  let(:supervisor) { create(:admin, :manager, :with_access, resource: organization) }
+  let(:facility_group) { create(:facility_group, organization: organization) }
+  let!(:small_facility1) { create(:facility, name: 'small1', facility_size: 'small', facility_group: facility_group) }
+  let!(:small_facility2) { create(:facility, name: 'small2', facility_size: 'small', facility_group: facility_group) }
+  let!(:medium_facility1) { create(:facility, name: 'medium1', facility_size: 'medium', facility_group: facility_group) }
+  let!(:medium_facility2) { create(:facility, name: 'medium2', facility_size: 'medium', facility_group: facility_group) }
+  let!(:large_facility1) { create(:facility, name: 'large1', facility_size: 'large', facility_group: facility_group) }
+  let!(:large_facility2) { create(:facility, name: 'large2', facility_size: 'large', facility_group: facility_group) }
+  let(:all_facilities) { [small_facility1, small_facility2, medium_facility1, medium_facility2, large_facility1, large_facility2] }
+  let(:period) { Period.month(today) }
```

Comment:
This is why the test data is getting returned - our Periods always start at the beginning of months (or quarters).  The middle of the month date is throwing things off. - 

Changing this to `let(:period) { Reports::RegionService.default_period }` should get you data coming back.